x-pack/plugins/ux/common/utils/merge_projection.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DeepPartial } from 'utility-types';
+import { cloneDeep, isPlainObject, mergeWith } from 'lodash';
+
+type PlainObject = Record<string | number | symbol, any>;
+
+type SourceProjection = DeepPartial<any>;
+
+type DeepMerge<T, U> = U extends PlainObject
+  ? T extends PlainObject
+    ? Omit<T, keyof U> & {
+        [key in keyof U]: T extends { [k in key]: any }
+          ? DeepMerge<T[key], U[key]>
+          : U[key];
+      }
+    : U
+  : U;
+
+export function mergeProjection<T extends any, U extends SourceProjection>(
```

Comment:
Ended up moving this over because there's logic about projection in the query generator for this component. LMK if this can be somehow skipped and we can delete this file.

Reply:
APM doesn't use projections anymore. If you also want to get rid of projections for the UX related code you are welcome to do it as part of this PR or a follow-up. If you do that we can delete the `merge_projections.ts` in both APM and UX.

Reply:
@sqren I think there are one or two other components in APM relying on that code. When I take them out I'll make sure nothing else references the projections and delete then.

@shahzad31 I'd appreciate your thoughts on whether we want this projection code to move into UX or if we can remove it here and now.

Reply:
i think if it makes sense we can move these. If you think the util apm is using in place of projections, it's also worth exploring those.